src/bzadmin/ServerLink.cxx

```diff
@@ -541,7 +541,7 @@ int ServerLink::read(uint16_t& code, uint16_t& len, void* msg, int blockTime)
     // block for specified period.  default is no blocking (polling)
     struct timeval timeout;
     timeout.tv_sec = blockTime / 1000;
-    timeout.tv_usec = blockTime - 1000 * timeout.tv_sec;
+    timeout.tv_usec = 1000 * (blockTime % 1000);
```

Comment:
This is more correct than what was there, presuming that `blockTime` is actually provided in milliseconds. Unfortunately, there's not enough context here to know.

Reply:
It is actually in milliseconds as far as I can tell. There's a comment in [src/bzadmin/BZAdminClient.cxx:150](https://github.com/BZFlag-Dev/bzflag/blob/e55a68ce79084864dca2ff7ec8565b022a640a32/src/bzadmin/BZAdminClient.cxx#L150) referring to this.
```
// read until we have a package, or until we have waited 100 ms
if (sLink.read(code, len, inbuf, 100) == 1)
```

Additionally, the calculation `timeout.tv_sec = blockTime / 1000;` also suggest that the blockTime is milliseconds.